### PR TITLE
Perform ISO rendering using a background queue

### PR DESCRIPTION
**Pull request by jayzes**
Instead of our current thread worker implementation. This should theoretically give us the ability to scale iso rendering up/down based on demand, and not have it be tied to the limitations of web worker processes.

Uses https://github.com/Automattic/kue since it has a nice job status API.

TODO:

- [x] Implement queue worker side of things (should just be modding `server-render-entrypoint.js`)
- [x] Add queue worker to Procfile
- [x] Figure out queue length monitoring
- [x] Figure out process concurrency and sizing
- [ ] Add/update tests

Deployment (ninja/epic):
- [ ] Add Redis plan to `ninja`
- [ ] Set `REDIS_PROVIDER` env var
- [ ] Set `CLUSTER_WORKERS` env var (depends on dyno size - maybe 2 for a 2x instance?)
- [ ] Deploy!
- [ ] Scale `queue` worker to 1 2x dyno
- [ ] Scale down web worker to 1 2x dyno
- [ ] Smoke test
- [ ] Add Redis plan to `epic`
- [ ] Set `REDIS_PROVIDER` env var
- [ ] Set `CLUSTER_WORKERS` env var (depends on dyno size - maybe 10 for a PM/PL instance?)
- [ ] Deploy!
- [ ] Scale `queue` worker to 1-2 PM/PL dynos
- [ ] Scale down web worker to 1-2 2x dynos
- [ ] Roll config to `stage1`/`stage2`/`rainbow`
- [ ] Add Librato alerts as needed

### CODE REVIEW

File: src/prerender.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        done(null, { type: 'render', body: html })
+      }
+    }).catch((err) => {
+      // this will give you a js error like:
+      // `window is not defined`
+      console.log('ELLO CATCH ERROR', err)
+      Honeybadger.notify(err);
+      done(null, { type: 'error' })
+    })
+    renderToString(InitialComponent)
```
Comment (jayzes) [inline src/prerender.js]: I think this line might might be doubling our work unnecessarily as we call the same thing on line 83

**Comment by ellodanger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/prerender.js (hunk 1 above)
Comment (mkitt) [inline src/prerender.js]: Yikes.. sort of looks that way. 💰

Comment (jayzes) [inline src/prerender.js]: I'm now not 100% as I just found https://github.com/redux-saga/redux-saga/issues/304 and https://github.com/redux-saga/redux-saga/issues/418 - probably a question for @rynbyjn. I don't understand sagas well enough to understand what's going on there.

**Comment by jayzes** [comment]
Found this as well: https://gist.github.com/sompylasar/5e7157e451f4b7268def9ae1ce01edd4#file-serverrenderloop-js

**Comment by jayzes** [comment]
@rynbyjn that last commit implements https://www.pivotaltracker.com/story/show/134092451

File: src/server-iso.js [hunk 2]
```diff
@@ -80,71 +79,66 @@ function saveResponseToCache(cacheKey, body) {
...
-        // Abnormal exit, may be in a dirty state
-        if (code !== 0) {
-          console.log(`- Render process exited with ${code} due to ${signal}`)
+
+    // Kick off the render
+    const renderOpts = {
+      access_token: token.token.access_token,
+      originalUrl: req.originalUrl,
+      url: req.url,
+      timingHeader: timingHeader
```
Comment (rynbyjn) [inline src/server-iso.js]: This is the CI failure.. should just be `timingHeader,`

File: src/selectors/authentication.js [hunk 3]
```diff
@@ -8,14 +8,12 @@ export const selectRefreshToken = state => state.authentication.get('refreshToke
 export const selectShouldUseAccessToken = (state) => {
   const accessToken = selectAccessToken(state)
   const expDate = selectExpirationDate(state)
-  const isLoggedIn = selectIsLoggedIn(state)
-  return isLoggedIn && accessToken && expDate > new Date()
+  return accessToken && expDate > new Date()
```
Comment (rynbyjn) [inline src/selectors/authentication.js]: Makes sense and shouldn't effect anything else that I can think of 🤘

**Comment by jayzes** [comment]
Okay, think this is ready to go. Adjusted the one bug that @rynbyjn found, corrected the refresh token behavior, cleaned up commits, rebased from master, and re-deployed to rainbow for another set of tests.

**Pull request merged**